src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Authentication;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up cookie authentication services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class CookieAuthenticationServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds cookie authentication services to the specified <see cref="IServiceCollection" />. 
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public static IServiceCollection AddCookieAuthentication(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddAuthentication();
+            services.AddAntiforgery();
+            return services;
+        }
+
+        /// <summary>
+        /// Adds cookie authentication services to the specified <see cref="IServiceCollection" />. 
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="configureOptions">An action delegate to configure the provided <see cref="SharedAuthenticationOptions"/>.</param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, Action<SharedAuthenticationOptions> configureOptions)
```

Comment:
Yeah, it's a little weird, but it's more correct than using CookieAuthenticationOptions. CookieAuthenticationOptions can't be in DI because there are commonly multiple instances.


I think we should just drop the parameter. Typically cookies doesn't use the shared options at all so I don't think there's any real loss of usage by requiring an explicit configure if they wanted to configure the shared.